demo/site/src/middleware/domainRewrite.ts

```diff
@@ -1,8 +1,23 @@
+import { previewParams, SitePreviewData } from "@comet/cms-site";
 import { getHostByHeaders, getSiteConfigForHost } from "@src/util/siteConfig";
 import { NextRequest, NextResponse } from "next/server";
 
 import { CustomMiddleware } from "./chain";
 
+function mapPreviewDataToPreviewParam(previewData: SitePreviewData | undefined) {
+    if (previewData) {
+        if (previewData.includeInvisible) return "blocks";
+        return "pages";
+    }
```

Comment:
The differentiation between pages (= site preview?) and blocks (= block preview?) could be confusing because the site preview supports invisible content as well.